### DISCUSSION TITLE
How can I disable AutoScroll if the user is not at the end of the chart?

### DISCUSSION THREAD

**Discussion by peshkira** [patrykandpatrick/vico#761] [Questions]
### Question

Hi!

I am using Vico 2.0.0.alpha21 with Compose-M3 and I noticed that you removed the Scroll Listener in the alpha09.
My chart displays time-based entries and always auto-scrolls to the end. Currently I use a custom AutoScrollCondition that just compares if there is a change in the old and new model size and if there is, it returns true. (Very similar to the one that you provide, but not only for larger models, but for any change).

I would like to now implement a custom auto scroll condition, which would auto scroll, only if the user sees the last point in the chart (at the maxX) , but not otherwise. In other words, if the user scrolls back in time, the auto scroll is disabled and new data is appended, without the user noticing, but once they scroll back to the end (maxX), I want the autoscroll to continue automatically.

I hope I am explaining it correct. I had a similar implementation with an older version of the library, but there I had access to the ScrollListener. Now I am unsure how to achieve this.

Thank you very much.

### Vico version(s)

2.0.0.alpha21

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, please see [here](https://patrykandpatrick.com/vico/releases/2.0.0-alpha.9/#scroll).
> Note that listeners have been removed—read the current and maximum scroll values via the value and maxValue fields, which use MutableFloatState.

**Reply by peshkira** [reply]
Hey!

Thanks for replying, I saw this actually, but for some reason couldn't make it work. Will give it another go, but would appreciate an example if you have one somewhere.

Best

**Reply by peshkira** [reply]
Alright!

I was able to do it with a snapshotFlow. I am not sure this is how you intended it, but it seems to work.

```
LaunchedEffect(scrollState) {
  snapshotFlow { scrollState.value }
      .distinctUntilChanged()
      .collect {
          println("/// scroll position: $it")
      }
  }
```

Thanks again for your work and the amazing lib. 
cheers.

**Reply by Gowsky** [reply]
Yes, it looks good.
